### PR TITLE
Add Faster R-CNN VOC07+12 experiment

### PR DESCRIPTION
**Pull request by yuyu2172**
Merge after #265

### CODE REVIEW

File: chainercv/links/model/faster_rcnn/faster_rcnn_vgg.py [hunk 1]
```diff
@@ -75,7 +75,12 @@ class FasterRCNNVGG16(FasterRCNN):
             'url': 'https://github.com/yuyu2172/share-weights/releases/'
             'download/0.0.4/'
             'faster_rcnn_vgg16_voc07_trained_2017_08_06.npz'
-        }
+        },
+        'voc0712': {
+            'n_fg_class': 20,
+            'url': 'https://github.com/yuyu2172/share-weights/releases/'
+            'download/0.0.4/faster_rcnn_vgg16_voc0712_2017_07_21.npz'
```
Comment (Hakuyume) [inline chainercv/links/model/faster_rcnn/faster_rcnn_vgg.py]: This model is not marked as `_trained_` although VOC07 model is marked. Was this model converted from Caffe or other frameworks?

File: examples/faster_rcnn/README.md [hunk 2]
```diff
@@ -44,11 +45,14 @@ On top of that, the anchors are not discretized in ChainerCV.
...
+```
+$ python train.py --dataset voc07 --gpu GPU --step_size 50000  --iteration 70000
+```
 
-You can train the model with the following code.
-
+For training with VOC07+12
 ```
-$ python train.py [--gpu <gpu>]
+$ python train.py --dataset voc0712 --gpu GPU --step_size 80000  --iteration 110000
```
Comment (Hakuyume) [inline examples/faster_rcnn/README.md]: `$ python train.py --dataset voc0712 --step_size 80000  --iteration 110000 [--gpu <gpu>]`

File: examples/faster_rcnn/README.md [hunk 3]
```diff
@@ -44,11 +45,14 @@ On top of that, the anchors are not discretized in ChainerCV.
 
 
 ### Train code
+For training with VOC07
+```
+$ python train.py --dataset voc07 --gpu GPU --step_size 50000  --iteration 70000
```
Comment (Hakuyume) [inline examples/faster_rcnn/README.md]: `$ python train.py --dataset voc07 --step_size 50000 --iteration 70000 [--gpu <gpu>]`

File: examples/faster_rcnn/README.md [hunk 4]
```diff
@@ -4,7 +4,8 @@
 
 | Training Setting | Evaluation | Reference | ChainerCV |
 |:-:|:-:|:-:|:-:|
-| VOC 2007 trainval | VOC 2007 test|  69.9 mAP [1] | 70.6 mAP |
+| VOC07 trainval | VOC07 test|  69.9 mAP [1] | 70.6 mAP |
+| VOC2007\&2012 trainval | VOC07 test|  73.2 mAP [1] | 74.7 mAP |
```
Comment (Hakuyume) [inline examples/faster_rcnn/README.md]: I think adding spaces around `|` is beautiful. `VOC07 test|` -> `VOC07 test |`

Comment (Hakuyume) [reply examples/faster_rcnn/README.md]: `VOC07 trainval` and `VOC2007\&2012 trainval` look inconsistent.

Comment (yuyu2172) [reply examples/faster_rcnn/README.md]: Thank you for pointing that out.

**Pull request merged**